### PR TITLE
Lesson 5 Exercises

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: src/chapters/unit-testing/exercises.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+   .. sourcecode:: csharp
+
+      //TODO: can't have more gas than tank size, expect an exception
+      [TestMethod]
+      public void TestGasOverfillException() {
+
+      }
+
+#. Now we need to tell MSTest the test passes if an exception is thrown. We will use a new annotation ``[ExpectedException]``.
```
Comment (CarlyLanglois) [inline src/chapters/unit-testing/exercises.rst]: change annotation to attribute

File: src/chapters/unit-testing/exercises.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+#. Update the ``Car`` class to include an ``AddGas()`` method.
+
+   .. sourcecode:: csharp
+
+      public void AddGas(double gas) {
+        GasTankLevel += gas;
+      }
+
+#. Back in ``CarTest``, implement the new ``AddGas()`` method and a
```
Comment (CarlyLanglois) [inline src/chapters/unit-testing/exercises.rst]: CarTests*

File: src/chapters/unit-testing/exercises.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+=======================
+
+Work on these exercises in your local copy of the `csharp-web-dev-lsn5unittesting <https://github.com/LaunchCodeEducation/csharp-web-dev-lsn5unittesting>`_ repo.
+You were directed to fork and clone this repo earlier in the chapter. If you have not done so, do so now!
+Before starting work on the exercises, check out the ``exercises-start`` branch.
+
+.. admonition:: Tip
+
+   Earlier in the chapter, we instantiated an object of the ``Car`` class called ``test_car``.
+   Be sure to take note of the values of the ``make``, ``model``, ``GasTankSize``, and ``MilesPerGallon`` properties!
```
Comment (CarlyLanglois) [inline src/chapters/unit-testing/exercises.rst]: this refers to the fields so use lowercase for these items

File: src/chapters/unit-testing/exercises.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+   .. sourcecode:: csharp
+
+      test_car.Drive(50);
+
+#. With a value of ``50`` miles passed into ``Drive()``, we expect 
+   ``test_car`` to have a ``GasTankLevel`` of ``9``.
+
+   .. sourcecode:: csharp
+
+      Assert.AreEqual(9, test_car.GasTankLevel(), .001);
```
Comment (CarlyLanglois) [inline src/chapters/unit-testing/exercises.rst]: test_car.GasTankLevel

File: src/chapters/unit-testing/exercises.rst [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+``TestGasTankAfterExceedingTankRange()``
+----------------------------------------
+
+Add a test for the fourth TODO, "GasTankLevel is accurate after attempting to drive past tank range".
+
+#. You're on your own for this one. You'll need to simulate the ``Car``
+   travelling farther than it's ``GasTankLevel`` allows.
```
Comment (CarlyLanglois) [inline src/chapters/unit-testing/exercises.rst]: farther than the ``gasTankLevel`` value allows.

File: src/chapters/unit-testing/exercises.rst [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+      test_car.AddGas(5);
+      Assert.Fail("Shouldn't get here, car cannot have more gas in tank than the size of the tank");
+
+#. Run the test. It should fail! In the output log, we can see our 
+   ``Assert.Fail()`` statement about not being able to add more gas printed out.
+
+#. We need to refactor ``Car`` to throw an exception when too much
+   gas is added to the tank. Find the ``AddGas()`` method and
+   modify it by adding the following code in the appropriate place
```
Comment (CarlyLanglois) [inline src/chapters/unit-testing/exercises.rst]: place.

**Review by CarlyLanglois — APPROVED** [review]

File: src/chapters/unit-testing/exercises.rst (hunk 1 above)
Comment (gildedgardenia) [reply src/chapters/unit-testing/exercises.rst]: Done!

File: src/chapters/unit-testing/exercises.rst (hunk 2 above)
Comment (gildedgardenia) [reply src/chapters/unit-testing/exercises.rst]: Done!

File: src/chapters/unit-testing/exercises.rst (hunk 3 above)
Comment (gildedgardenia) [reply src/chapters/unit-testing/exercises.rst]: Done!

File: src/chapters/unit-testing/exercises.rst (hunk 4 above)
Comment (gildedgardenia) [reply src/chapters/unit-testing/exercises.rst]: Done!

File: src/chapters/unit-testing/exercises.rst (hunk 5 above)
Comment (gildedgardenia) [reply src/chapters/unit-testing/exercises.rst]: Done!

File: src/chapters/unit-testing/exercises.rst (hunk 6 above)
Comment (gildedgardenia) [reply src/chapters/unit-testing/exercises.rst]: Done!

**Pull request merged**